### PR TITLE
docs: add fragment to native-components-android.md

### PR DESCRIPTION
**Pull request by f4z3k4s**
As a Javascript / React-Native developer I had a hard time figuring out how to integrate a custom native SDK with React Native using Android Fragments. Please correct anything that you wish. This information took me 3 days to figure out, especially the part with `setupLayout` and `manuallyLayoutChildren` so I wish an example like this was available in the official documentation. And I think I am not alone. I was expecting the code without `setupLayout` and `manuallyLayoutChildren` to work but i had no luck. If somebody has any better solution in your team, or any explanation, it would also be really nice to state in the documentation. Thank you for React Native, I love it.

### CODE REVIEW

File: docs/native-components-android.md [hunk 1]
```diff
@@ -188,3 +188,234 @@ MyCustomView.propTypes = {
...
+# Integration with an Android Fragment Example
+
+In order to integrate existing Native UI elements to your React Native app, you might need to use Android Fragments to give you a more granular control over your native component than returning a `View` from your `ViewManager`. You will need this if you want to add custom logic that is tied to your view with the help of [lifecycle methods](https://developer.android.com/guide/fragments/lifecycle), such as `onViewCreated`, `onPause`, `onResume`. The following steps will show you how to do it:
+
+## 1. Create a `Fragment`
+
+`MyFragment.java`
+
+```java
+public class MyFragment extends Fragment {
```
Comment (cortinico) [inline docs/native-components-android.md:201]: Adding an import on the `androidx` version of `Fragment` would be beneficial.

File: docs/native-components-android.md [hunk 2]
```diff
@@ -188,3 +188,234 @@ MyCustomView.propTypes = {
...
+    }
+}
+```
+
+## 2. Create the `ViewManager` subclass
+
+`MyViewManager.java`
+
+```java
+ public class MyViewManager extends ViewGroupManager<FrameLayout> {
```
Comment (cortinico) [inline docs/native-components-android.md:246]: The is an extra space at the beginning of this line

File: docs/native-components-android.md [hunk 3]
```diff
@@ -188,3 +188,234 @@ MyCustomView.propTypes = {
...
+        // e.g.: customView.onDestroy();
+    }
+}
+```
+
+## 2. Create the `ViewManager` subclass
+
+`MyViewManager.java`
+
+```java
```
Comment (cortinico) [inline docs/native-components-android.md:245]: I would add imports here. Specifically the one on `com.facebook.react.uimanager.ViewGroupManager`.

File: docs/native-components-android.md [hunk 4]
```diff
@@ -188,3 +188,234 @@ MyCustomView.propTypes = {
...
+      default: {}
+    }
+  }
+
+  /**
+   * Replace your React Native view with a custom fragment
+   */
+  public void createFragment(FrameLayout root, int reactNativeViewId) {
+    View parentView = (ViewGroup) root.findViewById(reactNativeViewId).getParent();
+    setupLayout((ViewGroup) parentView);
```
Comment (cortinico) [inline docs/native-components-android.md:301]: The double cast to `ViewGroup` is probably not needed. You should be fine with:
```suggestion
    ViewGroup parentView = (ViewGroup) root.findViewById(reactNativeViewId).getParent();
    setupLayout(parentView);
```

File: docs/native-components-android.md [hunk 5]
```diff
@@ -188,3 +188,234 @@ MyCustomView.propTypes = {
...
+    FragmentActivity activity = (FragmentActivity) reactContext.getCurrentActivity();
+    activity.getSupportFragmentManager()
+            .beginTransaction()
+            .replace(reactNativeViewId, myFragment, String.valueOf(reactNativeViewId))
+            .commit();
+  }
+
+  /**
+   * Setup layout
+   */
```
Comment (cortinico) [inline docs/native-components-android.md:313]: This can be removed

**Review by cortinico — COMMENTED** [review]
Great writeup @f4z3k4s! Sorry for the late review but we're going through our backlog of opened PRs. 

Please let me know if you're able to address the comments I left or not as I think we should merge this.

**Comment by f4z3k4s** [comment]
@cortinico Thanks for the review. I hopefully fixed everything you asked.

**Review by cortinico — APPROVED** [review]
> @cortinico Thanks for the review. I hopefully fixed everything you asked.

Yup great stuff 👍

**Comment by wilsolutions** [comment]
It would be nice to include an example of the CustomView as well...
More details: https://github.com/facebook/react-native-website/issues/2838

**Comment by f4z3k4s** [comment]
@wilsolutions Good point. In my case, it was a `MapView` from Here Maps SDK. That `MapView` extends `FrameLayout`. I guess it has to be a `FrameLayout`, but I am not too experienced with Android.

**Comment by f4z3k4s** [comment]
Okay, I've found the issue. I'll post a PR fixing this. Thanks @wilsolutions for spotting it. I'll also add a sample for CustomView.

**Comment by f4z3k4s** [comment]
Fix is in: https://github.com/facebook/react-native-website/pull/2843. @cortinico please have a look when you have time, it will help people in the future.

**Pull request merged**